### DISCUSSION TITLE
Installation in Eclipse 2024-12 not working

### DISCUSSION THREAD

**Discussion by mrodriguezyerro** [Genuitec/Copilot4Eclipse-community#60] [Help]
I'm unable to install Copilo4Eclispe in Eclipse 2024-12.

**Steps**

1. Downloaded [Eclipse IDE for Enterprise Java and Web Developers](https://www.eclipse.org/downloads/packages/release/2024-12/r/eclipse-ide-enterprise-java-and-web-developers)
2. Unzipped
3. Tried to install from Eclipse Marketplace
4. Tried to install from Install new software (with Contact all update sites... checked)

**Errors**

_Eclipse Marketplace_
An error occurred while collecting items to be installed
session context was:(profile=epp.package.jee, phase=org.eclipse.equinox.internal.p2.engine.phases.Collect, operand=, action=).
Can't download artifact osgi.bundle,com.genuitec.copilot4eclipse,1.3.2.202411110916 required by com.genuitec.copilot4eclipse[1.3.2.202411110916], org.apache.httpcomponents.httpcore[4.4.16.v20221207-1049], org.eclipse.wildwebdeveloper.embedder.node[1.0.3.202309272022], com.genuitec.copilot4eclipse.branding[1.3.2.202411110916], com.genuitec.copilot4eclipse.chat[1.3.2.202411110916], com.genuitec.copilot4eclipse.utils[1.3.2.202411110916], org.apache.commons.logging[1.2.0.v20180409-1502], com.genuitec.copilot4eclipse.licaware[1.3.2.202411110916], com.genuitec.copilot4eclipse.core.feature.feature.group[1.3.2.202411110916], org.apache.commons.jxpath[1.3.0.v200911051830], com.genuitec.copilot4eclipse[1.3.2.202411110916], javax.xml[1.3.4.v201005080400] from any of the following repositories: 
https://www.genuitec.com/updates/copilot4eclipse
https://download.eclipse.org/releases/2024-03

_Install new software_
An error occurred while collecting items to be installed
session context was:(profile=epp.package.jee, phase=org.eclipse.equinox.internal.p2.engine.phases.Collect, operand=, action=).
Can't download artifact osgi.bundle,com.genuitec.copilot4eclipse,1.3.2.202411110916 required by com.genuitec.copilot4eclipse[1.3.2.202411110916], org.apache.httpcomponents.httpcore[4.4.16.v20221207-1049], org.eclipse.wildwebdeveloper.embedder.node[1.0.3.202309272022], com.genuitec.copilot4eclipse.branding[1.3.2.202411110916], com.genuitec.copilot4eclipse.chat[1.3.2.202411110916], com.genuitec.copilot4eclipse.utils[1.3.2.202411110916], org.apache.commons.logging[1.2.0.v20180409-1502], com.genuitec.copilot4eclipse.licaware[1.3.2.202411110916], org.eclipse.lsp4e[0.18.14.202410030435], com.genuitec.copilot4eclipse.core.feature.feature.group[1.3.2.202411110916], org.apache.commons.jxpath[1.3.0.v200911051830], com.genuitec.copilot4eclipse[1.3.2.202411110916], javax.xml[1.3.4.v201005080400] from any of the following repositories: 
file:/C:/Work/Development/Eclipse/jee-2024-12/configuration/org.eclipse.osgi/402/data/listener_1925729951/
file:/C:/Work/Development/Eclipse/jee-2024-12/
file:/C:/Work/Development/Eclipse/jee-2024-12/p2/org.eclipse.equinox.p2.core/cache/
https://download.eclipse.org/eclipse/updates/4.33
https://download.eclipse.org/mylyn/updates/release/4.5.0
https://download.eclipse.org/tools/orbit/simrel/orbit-aggregation/release/4.33.0
https://download.eclipse.org/lsp4j/updates/releases/0.23.1
http://download.eclipse.org/usssdk/updates/release/latest
https://download.eclipse.org/technology/m2e/releases/latest
https://download.eclipse.org/lsp4e/releases/0.26.6
https://download.eclipse.org/releases/2024-12
https://download.eclipse.org/webtools/downloads/drops/R3.35.0/R-3.35.0-20240825085431/repository
https://download.eclipse.org/webtools/repository/latest
https://download.eclipse.org/technology/epp/packages/latest/
https://download.eclipse.org/releases/latest
https://download.eclipse.org/eclipse/updates/4.34
https://download.eclipse.org/justj/epp/release/latest
https://download.eclipse.org/tm4e/releases/0.14.0
https://www.genuitec.com/updates/copilot4eclipse
https://download.eclipse.org/modeling/emf/emf/builds/release/2.40.0
https://download.eclipse.org/releases/2024-03
https://download.eclipse.org/egit/updates-7.1
https://download.eclipse.org/wildwebdeveloper/releases/1.3.9
https://download.eclipse.org/justj/jres/21/updates/release/latest

**Comment by wayneparrott** [comment]
Thanks for sharing the details of your failed installation. Just now I attempted to replicate the issue you observed and am able to successfully  install Copilot4Eclipse in the new Eclipse 2024-12 Jee version. 

Q1: Have you successfully installed Copilot4Eclipse previously with earlier versions of Eclipse?
I'm wondering if there was a glitch with our network during the time you attempted to install Copilot4Eclipse. 
Please try installing Copilot4Eclipse again from the marketplace. 

Q2: If you are still having an installation error, are you operating behind a corporate network proxy or firewall?
If yes please check your Eclipse network settings. If the problem still persists, consult your network team to learn if installing plugins from sites other than downloads.eclipse.org is allowed.

Let us know if you continue to experience this installation issue.

Wayne
Copilot4Eclipse Team

**Comment by mrodriguezyerro** [comment]
Reading your message reminded me that it might be necessary to configure the default Eclipse JVM cacerts. I did that, but I still get the same error.

Regarding whether I had installed it before, yes, I had installed a previous version in Eclipse 2024-06.

Today I installed other plugins (JBoss Tools, SonarQube for IDE 10.9.1...) without any issues.

I just tested between 13:39 and 14:05 Spanish time. 

Last error:
> eclipse.buildId=4.34.0.20241128-0756
> java.version=21.0.5
> java.vendor=Eclipse Adoptium
> BootLoader constants: OS=win32, ARCH=x86_64, WS=win32, NL=es_ES
> Framework arguments:  -product org.eclipse.epp.package.jee.product
> Command-line arguments:  -os win32 -ws win32 -arch x86_64 -product org.eclipse.epp.package.jee.product
> 
> org.eclipse.equinox.p2.engine
> Error
> Thu Dec 05 14:04:52 CET 2024
> Can't download artifact osgi.bundle,com.genuitec.copilot4eclipse,1.3.2.202411110916 required by com.genuitec.copilot4eclipse[1.3.2.202411110916], org.eclipse.wildwebdeveloper.embedder.node[1.0.3.202309272022], com.genuitec.copilot4eclipse.branding[1.3.2.202411110916], com.genuitec.copilot4eclipse.chat[1.3.2.202411110916], com.googlecode.javaewah.JavaEWAH[1.2.3], com.genuitec.copilot4eclipse.utils[1.3.2.202411110916], com.genuitec.copilot4eclipse.licaware[1.3.2.202411110916], com.genuitec.copilot4eclipse.core.feature.feature.group[1.3.2.202411110916], com.genuitec.copilot4eclipse[1.3.2.202411110916], org.apache.commons.commons-collections4[4.4.0], javax.xml[1.3.4.v201005080400] from any of the following repositories: 
> https://www.genuitec.com/updates/copilot4eclipse
> https://download.eclipse.org/releases/2024-03
>
